pkg/controller/beat/common/reconcile.go

```diff
@@ -71,13 +71,13 @@ func reconcilePodVehicle(podTemplate corev1.PodTemplateSpec, params DriverParams
 		results.WithError(err)
 	}
 
-	err = updateStatus(params, ready, desired)
-	if err != nil && apierrors.IsConflict(err) {
-		params.Logger.V(1).Info(
-			"Conflict while updating status",
+	err = calculateStatus(params, ready, desired, params.Status)
+	if err != nil {
+		params.Logger.Error(
+			err,
+			"While calculating new status",
```

Comment:
I would wrap the error to pass the context upwards and maybe not log here as controller runtime will log it anyway